jung-visualization/src/main/java/edu/uci/ics/jung/visualization/spatial/SpatialRTree.java

```diff
@@ -298,18 +296,18 @@ public void recalculate() {
     }
 
     @Override
-    public void layoutChanged(LayoutEvent<N> evt) {
-      this.update(evt.getNode(), evt.getLocation());
+    public void layoutNodePositionChanged(LayoutNodePositionChange.NetworkEvent<N> evt) {
+      update(evt.node, evt.location);
     }
 
     @Override
-    public void layoutChanged(LayoutNetworkEvent<N> evt) {
-      update(evt.getNode(), evt.getLocation());
+    public void layoutNodePositionChanged(LayoutNodePositionChange.Event<N> evt) {
+      update(evt.node, evt.location);
     }
   }
 
   public static class Edges<E, N> extends SpatialRTree<E, N>
-      implements Spatial<E>, LayoutChangeListener<N> {
```

Comment:
not sure if it was clear why this is <E, N>.
The element shapes stored in the tree are the edges, so the rtree is a collection of E shapes (as opposed to N shapes). In order to work with the edges, the tree also needs access to the Node endpoints. Thus, the SpatialRTree.Edges<E, N> implements Spatial<E>
(an RTree of E shapes with a layout model of N things)
and
SpatialRTree.Nodes<N, N> implements Spatial<N>
(an RTree of N shapes with a layoutMode of N things)